### PR TITLE
Update lists status subscription from ActiveCampaign (#44)

### LINKED ISSUE #44: Add webhook for list subscription status from AC

### PR DESCRIPTION
**Pull request by lruozzi9**
Fixes #44 

Changes proposed in this pull request:
- Added update of lists status subscription from ActiveCampaign thanks to the AC's webhooks

@webgriffe/wg-devs what do you think about this PR :shipit:?

### CODE REVIEW

File: tests/Integration/Command/EnqueueWebhookCommandTest.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            '--all' => true,
+        ]);
+        self::assertEquals(0, $commandTester->getStatusCode());
+
+        $fashionChannel = $this->channelRepository->findOneBy(['code' => 'fashion_shop']);
+        $digitalChannel = $this->channelRepository->findOneBy(['code' => 'digital_shop']);
+        $transport = self::getContainer()->get('messenger.transport.main');
+        /** @var Envelope[] $messages */
+        $messages = $transport->get();
+        $this->assertCount(2, $messages);
```
Comment (LucaGallinari) [inline tests/Integration/Command/EnqueueWebhookCommandTest.php:88]: Why it didn't enqueue "other_shop" channel?

**Review by LucaGallinari — CHANGES_REQUESTED** [review]
The command that creates the webhooks is cool

File: src/Controller/WebhookController.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+{
+    public function __construct(
+        private ActiveCampaignCustomerRepositoryInterface $customerRepository,
+        private MessageBusInterface $messageBus
+    ) {
+    }
+
+    public function updateListStatusAction(Request $request): Response
+    {
+        $type = $request->request->getAlpha('type');
```
[the remark this answers is not in the document]
Comment (lruozzi9) [reply src/Controller/WebhookController.php:27]: ?

File: src/Command/EnqueueWebhookCommand.php [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    }
+
+    /**
+     * @param string|int|null $channelId
+     *
+     * @return string|int
+     */
+    public function validateChannelId($channelId)
+    {
+        if ($channelId === null || $channelId === '') {
```
[the remark this answers is not in the document]
Comment (lruozzi9) [reply src/Command/EnqueueWebhookCommand.php:160]: Phpstan says: `Construct empty() is not allowed. Use more strict comparison`.

File: tests/Integration/Command/EnqueueWebhookCommandTest.php (hunk 1 above)
Comment (lruozzi9) [reply tests/Integration/Command/EnqueueWebhookCommandTest.php:88]: All the channels have the same list id associated, but the other_shop channel won't be enqueued 'cause there is already a [webhook for that route and that list](https://github.com/webgriffe/SyliusActiveCampaignPlugin/pull/52/files#diff-bffa57b3c1d29ca765e7fc7d33f4839a92a1d68bad9624f3111b42a31085bedeR25).

File: src/Command/EnqueueWebhookCommand.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (LucaGallinari) [reply src/Command/EnqueueWebhookCommand.php:160]: Didn't know!

File: tests/Integration/Command/EnqueueWebhookCommandTest.php (hunk 1 above)
Comment (LucaGallinari) [reply tests/Integration/Command/EnqueueWebhookCommandTest.php:88]: Ohh ok, missed it. Good

**Review by LucaGallinari — APPROVED** [review]

**Pull request merged**